MFaaP.MFWSClient/MFWSVaultObjectSearchOperations.cs

```diff
@@ -148,12 +179,13 @@ public async Task<ObjectVersion[]> SearchForObjectsByConditionsAsync(Cancellatio
 		/// Searches the vault.
 		/// </summary>
 		/// <param name="searchConditions">The conditions to search for.</param>
+		/// <param name="limit">If 0, uses Mfiles default at server (default 500 items returned)</param>
 		/// <param name="token">A cancellation token for the request.</param>
 		/// <returns>An array of items that match the search conditions.</returns>
-		public ObjectVersion[] SearchForObjectsByConditions(CancellationToken token, params ISearchCondition[] searchConditions)
+		public ObjectVersion[] SearchForObjectsByConditions(CancellationToken token, int limit = 0, params ISearchCondition[] searchConditions)
```

Comment:
I recommend that the default value is not zero, as this is passed to [SearchForObjectsByConditionsEx](https://www.m-files.com/api/documentation/latest/index.html#MFilesAPI~VaultObjectSearchOperations~SearchForObjectsByConditionsEx.html) and zero means "unlimited".

Perhaps change to -1 to mean "default", as this is an invalid value anyway?

Reply:
solution in net PR